### PR TITLE
feat: Init command not using template properties's default value

### PR DESCRIPTION
**Pull request by nandorholozsnyak**
- Removing getter and setter pairs from TemplateProperty and using public modifier
- Applying template property defaults in init command by default, option to disable.
- Documentation for the new option

Solves this: https://github.com/jbangdev/jbang/issues/1231

### CODE REVIEW

**Comment by quintesse** [comment]
Personally I don't see much use for an option that doesn't apply default values. Do you @maxandersen ?

**Comment by maxandersen** [comment]
Yeah. Same. No need to be able to turn this off.

File: docs/modules/cli/pages/jbang-init.adoc [hunk 1]
```diff
@@ -65,6 +65,10 @@ Initialize a script.
 *--quiet*::
   jbang will be quiet, only print when error occurs.
 
+*-i*, *--ignore-template-defaults*=_<true|false>_::
```
Comment (maxandersen) [inline docs/modules/cli/pages/jbang-init.adoc:68]: I don't see value in having this flag. why would you not want the defaults applied?

File: src/main/java/dev/jbang/catalog/TemplateProperty.java [hunk 2]
```diff
@@ -9,31 +9,15 @@
  */
 public class TemplateProperty {
 
-	private String description;
+	public String description;
```
Comment (maxandersen) [inline src/main/java/dev/jbang/catalog/TemplateProperty.java:12]: why this change?

File: src/main/java/dev/jbang/cli/Init.java [hunk 3]
```diff
@@ -43,6 +44,10 @@
 			"--deps" }, converter = CommaSeparatedConverter.class, description = "Add additional dependencies (Use commas to separate them).")
 	List<String> dependencies;
 
+	@CommandLine.Option(names = { "-i",
```
Comment (maxandersen) [inline src/main/java/dev/jbang/cli/Init.java:47]: just remove it?

**Review by maxandersen — DISMISSED** [review]
I pushed update that removes the flag. if pass ci good to go.

File: src/main/java/dev/jbang/cli/Init.java [hunk 4]
```diff
@@ -109,6 +112,16 @@ public Integer doCall() throws IOException {
 		return EXIT_OK;
 	}
 
+	private void useTemplatePropertiesIfAllowed(dev.jbang.catalog.Template tpl) {
```
[the remark this answers is not in the document]
Comment (maxandersen) [reply src/main/java/dev/jbang/cli/Init.java]: fixed.

**Comment by nandorholozsnyak** [comment]
Sorry for not responding for the change requests, I'm on "holiday", thank you for the changes and merge.
